master_write.py

```diff
@@ -0,0 +1,103 @@
+
+                                        ## Now update temp_earliest
+                                        if(min(chi) <= bench_mark and gens < temp_earliest): # check if the min chi score is bellow threshold 
+                                            temp_earliest = gens
+                                        if(min(chi) <= bench_mark2 and gens < temp_earliest2):
+                                            temp_earliest2 = gens
+                                    temp_benchmark_gen.append(temp_earliest) 
+                                    temp_benchmark_gen2.append(temp_earliest2)
+
+                                    # Begin Plotting
+                                    plt.figure(count, figsize=(16,9))
+                                    plt.plot(gen_num, ave_chi, c=c, linestyle = 'dotted', label =('Run'+str(test)+'average'))
+                                    plt.plot(gen_num, min_chi, c=c, linestyle = 'solid', label = ('Run'+str(test)+'High'))
+                                    #plt.axis([0,50, 0.00, 1.5])
+                                    plt.grid(visible=True, which='major', color = '#666666', linestyle = '-', linewidth =0.5)
+                                    plt.minorticks_on()
+                                    plt.grid(visible=True, which = 'minor', color = '#999999', linestyle = '-', linewidth=0.2, alpha = 0.5)
+                                    plt.ylabel('Fitness Score')
+                                    plt.xlabel('Generations')
+                                    plt.suptitle(design +' Population:'+ str(population) +' Error:'+ error )
+                                    plt.savefig("plot" + design + error + str(rank) +'_'+ str(roulette) +'_'+ str(tournament) +'_'+ str(reproduction)+'_'+ str(crossover) +'_'+ str(mutation_rate) +'_'+ str(sigma)+'.png')
+                                plt.close()
+                                print("plot "+design+ str(rank) +'_'+ str(roulette) +'_'+ str(tournament) +'_'+ str(reproduction)+'_'+ str(crossover) +'_'+ str(mutation_rate) +'_'+ str(sigma) +" complete")
+                                earliest_gen.append(min(temp_benchmark_gen))
+                                average_gen.append(mean(temp_benchmark_gen))
+                                average_gen2.append(mean(temp_benchmark_gen2))
+                                sigma_average_gen.append(stat.pstdev(temp_benchmark_gen))
+                                sigma_average_gen2.append(stat.pstdev(temp_benchmark_gen2))
+
+
+
+
+with open("Master_gen.csv", 'w') as f:
+    f.write("Run Name, \t Earliest, \t Average (.5), \t Standard Deviation (.5), \t Average (.25), \t Standard Deviation (.25) \n")
+    for x in range(0, len(run_name)):
+        f.write(str(run_name[x]) + ", \t " + str(earliest_gen[x]) + ", \t " + str(average_gen[x]) + ", \t " + str(sigma_average_gen[x]) +", \t" +str(average_gen2[x]) + ", \t" + str(sigma_average_gen2[x]) + "\n")
+
+f.close()
+print( "Master_gen.csv written")
```

Comment:
```suggestion
with open("Master_gen.csv", 'w') as f:
    f.write("Run Name, \t Earliest, \t Average (.5), \t Standard Deviation (.5), \t Average (.25), \t Standard Deviation (.25) \n")
    for x in range(len(run_name)):
        f.write(str(run_name[x]) + ", \t " + str(earliest_gen[x]) + ", \t " + str(average_gen[x]) + ", \t " + str(sigma_average_gen[x]) + ", \t" +str(average_gen2[x]) + ", \t" + str(sigma_average_gen2[x]) + "\n")

print("Master_gen.csv written")
```
Why both commas and tabs?  Also why not using the csv module to write too?